### PR TITLE
Add table of contents to README 

### PR DESCRIPTION
**Pull request by wildmaples**
### Motivation

* Add table of contents to README 
* Link to `Migrating to Tapioca` on the README 

### Implementation
N/A

### Tests
N/A

### CODE REVIEW

**Review by paracycle — COMMENTED** [review]
Thank you for adding this but I am not sure if we need to maintain our own ToC when Github already adds a collapsed ToC for every Markdown file: https://github.blog/changelog/2021-04-13-table-of-contents-support-in-markdown-files/

This will be really hard to keep up to date as we add/remove things from the document, that's my biggest concern.

**Comment by st0012** [comment]
> when Github already adds a collapsed ToC for every Markdown file

Maybe it's just me, but I find it suboptimal compared with the document's own ToC because the extra click on a small icon. I know its existence the day it's released but I've only used it a handful of time.
But on the other hand, I think the usage section has too many headers. Perhaps we can ignore the 3rd level headers from the ToC? It should also make this easier to maintain.

**Comment by wildmaples** [comment]
Going on Stan's suggestion - is this any better? In any case, happy to remove the ToC completely, if not! 

Screen Shot 2022-07-14 at 12 42 10 PM

File: README.md [hunk 1]
```diff
@@ -30,6 +30,24 @@ Tapioca makes it easy to work with [Sorbet](https://sorbet.org) in your codebase
...
+  * [Excluding a gem from RBI generation](#excluding-a-gem-from-rbi-generation)
+  * [Changing the strictness level of the RBI for a gem](#changing-the-strictness-level-of-the-rbi-for-a-gem)
+  * [Keeping RBI files for gem up-to-date](#keeping-rbi-files-for-gems-up-to-date)
+  * [Pulling RBI annotations from remote sources](#pulling-rbi-annotations-from-remote-sources)
+  * [Generating RBI files for Rails and other DSLs](#generating-rbi-files-for-rails-and-other-dsls)
+  * [RBI files for missing constants and methods](#rbi-files-for-missing-constants-and-methods)
+  * [Configuration](#configuration)
+  * [Migrating to Tapioca](https://github.com/Shopify/tapioca/wiki/Migrating-to-Tapioca)
+* [Contributing](#contributing)
+* [License](#license)
```
Comment (Morriar) [inline README.md:49]: This should be generated or we're going to forget to update it.

We already have some code to do related things under https://github.com/Shopify/tapioca/blob/main/tasks/readme.rake. We could use a markdown parser to extract the  headings from the markdown.

**Comment by wildmaples** [comment]
Closing this in favour of https://github.com/Shopify/tapioca/pull/1068

**Pull request closed without merging**